src/viam/ur/module/utils.hpp

```diff
@@ -10,19 +12,18 @@ void configure_logger(const viam::sdk::ResourceConfig& cfg);
 
 // helper function to extract an attribute value from its key within a ResourceConfig
 template <class T>
-T find_config_attribute(const viam::sdk::ResourceConfig& cfg, const std::string& attribute) {
+std::optional<T> find_config_attribute(const viam::sdk::ResourceConfig& cfg, const std::string& attribute) {
     std::ostringstream buffer;
     auto key = cfg.attributes().find(attribute);
     if (key == cfg.attributes().end()) {
-        buffer << "required attribute `" << attribute << "` not found in configuration";
-        throw std::invalid_argument(buffer.str());
+        return std::nullopt;
     }
     const auto* const val = key->second.get<T>();
     if (!val) {
-        buffer << "required non-empty attribute `" << attribute << " could not be decoded";
+        buffer << "attribute `" << attribute << " could not be converted to the required type";
```

Comment:
You could move the decl of `buffer` into the body of the if, since it isn't used higher up anymore.